src/conn/conn_api.c

```diff
@@ -1739,8 +1739,8 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
     if (conn->is_new) {
         if (F_ISSET(conn, WT_CONN_READONLY))
             WT_ERR_MSG(session, EINVAL,
-              "Creating a new database is incompatible with "
-              "read-only configuration");
+              "The database metadata file is missing and a read-only connection cannot continue; "
+              "consider running recovery.");
```

Comment:
I don't feel the new message here is an improvement. The most common case is creating a new database, not that someone removed the turtle file or metadata file. At a minimum I suggest removing the "consider running recovery" because that is not helpful when creating a new database. @keithbostic and @agorrod - you were both involved more in the message discussion. If you prefer this, I'll go with it.

Reply:
I don't see how create is the likely/common path. You'd have to explicitly set the read-only flag on an operation that requires database creation, and that's not something someone is likely to do? What am I missing?

I don't care if we mention `metadata file is missing`, it feels like an internal detail, happy to see that go away.

I don't like error messages that guess at what the user is doing and so can point the user in the wrong direction as they try to fix the problem. (In other words, don't mention `create` unless we know without question that the user is trying to create a database.)

@luke-pearson, as far as I'm concerned, feel free to just decide without further input from me.